cartridge-cli.lua

```diff
@@ -2080,40 +2209,56 @@ local function pack_cpio(opts)
     info("Packing CPIO in: %s", pack_state.build_dir)
 
     local distribution_dir = fio.pathjoin(pack_state.build_dir, '/usr/share/tarantool/', pack_state.name)
-    form_distribution_dir(distribution_dir)
+    local ok, err = form_distribution_dir(distribution_dir)
+    if not ok then return false, err end
+
+    local ok, err = build_application(distribution_dir)
+    if not ok then return false, err end
+
+    local ok, err = generate_version_file(distribution_dir)
+    if not ok then return false, err end
 
-    build_application(distribution_dir)
-    generate_version_file(distribution_dir)
+    local ok, err = form_systemd_dir(pack_state.build_dir, opts)
+    if not ok then return false, err end
 
-    form_systemd_dir(pack_state.build_dir, opts)
-    write_tmpfiles_conf(pack_state.build_dir)
+    local ok, err = write_tmpfiles_conf(pack_state.build_dir)
+    if not ok then return false, err end
 
     if pack_state.tarantool_is_enterprise then
-        copy_taranool_binaries(distribution_dir)
+        local ok, err = copy_taranool_binaries(distribution_dir)
+        if not ok then return false, err end
     end
 
     local files = find_files(pack_state.build_dir, {include_dirs=true, exclude={'.git'}})
     files = filter_out_known_files(files)
 
-    write_file(fio.pathjoin(pack_state.build_dir, 'files'), table.concat(files, '\n'))
+    local ok, err = write_file(fio.pathjoin(pack_state.build_dir, 'files'), table.concat(files, '\n'))
+    if not ok then return false, err end
 
     local ok, pack_err = call("cd %s && cat files | %s -o -H newc > unpacked", pack_state.build_dir, cpio)
     if not ok then
-        die("Failed to pack CPIO: %s", pack_err)
+        return false, string.format("Failed to pack CPIO: %s", pack_err)
     end
 
     local payloadsize = fio.stat(fio.pathjoin(pack_state.build_dir, 'unpacked')).size
     local archive, read_err = check_output("cd %s && cat unpacked | %s -9", pack_state.build_dir, gzip)
     if archive == nil then
-        die("Failed to pack CPIO: %s", read_err)
+        return false, string.format("Failed to pack CPIO: %s", read_err)
     end
 
-    fio.unlink(fio.pathjoin(pack_state.build_dir, 'unpacked'))
-    fio.unlink(fio.pathjoin(pack_state.build_dir, 'files'))
+    for _, f in ipairs({'unpacked', 'files'}) do
+        local filepath = fio.pathjoin(pack_state.build_dir, f)
+        local ok, err = remove_by_path(filepath)
+        if not ok then return false, err end
+    end
 
     local fileinfo = generate_fileinfo(pack_state.build_dir)
 
-    return archive, fileinfo, payloadsize
+    return true, {
```

Comment:
I think in "positive" case we should return only one argument. Two arguments is sign of an error. Let's return `nil, err` in case of error and table otherwise

Reply:
Yes, thanks!